File: plugins/AzurePlugin/www/AzurePlugin.js
var exec = require('cordova/exec');

//开启通知
exports.resumePush = function(arg0, success, error) {
    exec(success, error, "AzurePlugin", "resumePush", [arg0]);
};
//关闭通知
exports.stopPush = function(arg0, success, error) {
    exec(success, error, "AzurePlugin", "stopPush", [arg0]);
};
//监听推送
exports.receiveNotificationInIOSCallback = function (data) {
  data = JSON.stringify(data)
  console.log('JPushPlugin:receiveNotificationInIOSCallback: ' + data)
  this.receiveNotification = JSON.parse(data)
  cordova.fireDocumentEvent('azure.receiveNotification', this.receiveNotification)
};

File: plugins/AzurePlugin/src/ios/AzurePlugin.m
//
//  AzurePlugin.m
//  HelloCordova
//
//  Created by wangsheng on 2017/4/6.
//
//

#import "AzurePlugin.h"
#import "HubInfo.h"
#import "AppDelegate+AzurePush.h"

@implementation AzurePlugin

-(void)pluginInitialize
{
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(didReceivedRemoteNotification:) name:kAZureNotification object:nil];
}


- (void)didReceivedRemoteNotification:(NSNotification *)notification
{
    NSDictionary *userInfo = notification.object;
    NSLog(@"userInfo:%@",userInfo);
    
    NSError *error;
    NSData *jsonStrData = [NSJSONSerialization dataWithJSONObject:userInfo  options:0 error:&error];
    NSString *jsonStr = [[NSString alloc] initWithData:jsonStrData encoding:NSUTF8StringEncoding];
    
    //收到通知
    dispatch_async(dispatch_get_main_queue(), ^{
        [self.commandDelegate evalJs:[NSString stringWithFormat:@"cordova.fireDocumentEvent('azure.receiveNotification',%@)",jsonStr]];
    });
}

-(void)stopPush:(CDVInvokedUrlCommand *)cmd
{
    [(AppDelegate *)[UIApplication sharedApplication].delegate unRegisterNotificationSetting];
}

-(void)resumePush:(CDVInvokedUrlCommand *)cmd
{
    [(AppDelegate *)[UIApplication sharedApplication].delegate registerNotificationSettting];
}

@end
